jerry-core/api/jerryscript.c

```diff
@@ -3249,27 +3249,11 @@ jerry_has_own_property (const jerry_value_t obj_val, /**< object value */
   ecma_object_t *obj_p = ecma_get_object_from_value (obj_val);
   ecma_string_t *prop_name_p = ecma_get_prop_name_from_value (prop_name_val);
 
-#if JERRY_BUILTIN_PROXY
-  if (ECMA_OBJECT_IS_PROXY (obj_p))
-  {
-    ecma_property_descriptor_t prop_desc;
-
-    ecma_value_t status = ecma_proxy_object_get_own_property_descriptor (obj_p, prop_name_p, &prop_desc);
-
-    if (ecma_is_value_true (status))
-    {
-      ecma_free_property_descriptor (&prop_desc);
-    }
-
-    return jerry_return (status);
-  }
-#endif /* JERRY_BUILTIN_PROXY */
-
-#if JERRY_BUILTIN_TYPEDARRAY
-  return jerry_return (ecma_op_ordinary_object_has_own_property (obj_p, prop_name_p));
-#else /* !JERRY_BUILTIN_TYPEDARRAY */
-  return ecma_op_ordinary_object_has_own_property (obj_p, prop_name_p);
-#endif /* JERRY_BUILTIN_TYPEDARRAY */
+#if JERRY_BUILTIN_PROXY || JERRY_BUILTIN_TYPEDARRAY
+  return jerry_return (ecma_op_object_has_own_property (obj_p, prop_name_p));
+#else /* !JERRY_BUILTIN_PROXY && !JERRY_BUILTIN_TYPEDARRAY */
+  return ecma_op_object_has_own_property (obj_p, prop_name_p);
+#endif /* JERRY_BUILTIN_PROXY || JERRY_BUILTIN_TYPEDARRAY */
```

Comment:
AFAIK we could simplify this to:
```c
return jerry_return(ecma_op_object_has_own_property(...));
```
I don't think the skipping of `jerry_return` call worth the extra #if-def guard.